tardis/utilities/executors/sshexecutor.py

```diff
@@ -3,43 +3,107 @@
 from ...interfaces.executor import Executor
 from ..attributedict import AttributeDict
 
+try:
+    from contextlib import asynccontextmanager
+except ImportError:
+    from aiotools import async_ctx_manager as asynccontextmanager
+
 import asyncssh
+import asyncio
+
+
+class SSHConnectionPool(object):
+    _connection_pool = dict()
+    _lock = None
+
+    def __init__(self, connection_pool_size=5, **parameters):
+        assert (
+            "host" in parameters
+        ), f"Host to connect to not found in parameters {parameters}"
+
+        self._connection_pool_size = connection_pool_size
+        self._host = parameters["host"]
+        self._parameters = parameters
+
+    @property
+    def acquire_lock(self):
```

Comment:
This naming seems misleading. The method doesn't acquire the lock, it just returns it.